Firestore/core/src/firebase/firestore/local/serializer.cc

```diff
@@ -140,6 +140,24 @@ std::unique_ptr<model::MaybeDocument> Serializer::DecodeMaybeDocument(
   }
 }
 
+void Serializer::EncodeDocument(Writer* writer, const model::Document& doc) const {
+  // Encode Document.name
+  writer->WriteTag({PB_WT_STRING, google_firestore_v1beta1_Document_name_tag});
+  writer->WriteString(rpc_serializer_.EncodeKey(doc.key()));
+
+  // Encode Document.fields (unless it's empty)
+  const ObjectValue& object_value = doc.data().object_value();
+  if (!object_value.internal_value.empty()) {
+    rpc_serializer_.EncodeObjectMap(writer, object_value.internal_value, google_firestore_v1beta1_Document_fields_tag, google_firestore_v1beta1_Document_FieldsEntry_key_tag, google_firestore_v1beta1_Document_FieldsEntry_value_tag);
+  }
+
+  // Encode Document.update_time
```

Comment:
Question: I presume `create_time` is deliberately ignored. If that's the case, perhaps (optionally) add a brief comment on why `update_time` is serialized but `create_time` isn't?

Reply:
The honest answer is because the objc implementation deliberately ignores it. I presume it's because we simply don't care about this field. (Or in other words, the on disk proto has "diverged", but not by enough for us to care.) I'll add a comment to that effect; Gil can correct me if I'm wrong.

Reply:
TL;DR: omitting creation time is the right call.

As usual, there's history: `create_time` is a read-only property and it's an error to send it to the backend so the remote serializer must omit it. The Objective-C local serializer implementation evolved out of the remote one and preserved this bias, though the bias is unnecessary.

Fixing this has never been a priority because we have no use for creation time and if we ever do need it we're going to need to deal with the on-device cache not having it when we do. I anticipate never needing this value so we can defer this indefinitely.